### PR TITLE
Implement ImageMagick pixel enumeration parser

### LINKED ISSUE #216: ImageMagick pixel enumeration parser
Is part of #41

Maybe it's gonna be a utility or something

### PR DESCRIPTION
**Pull request by rexim**
Close #216

### CODE REVIEW

**Comment by rexim** [comment]
@rexim also don't forget to check that probably a dash.el bug that you found on the stream

File: sprites/Makefile [hunk 1]
```diff
@@ -1,10 +1,14 @@
-PILL=Pill.json Pill.xpm
+TXTS=Big-Pill.txt Blinking-Terrified-Ghost.txt Pacman-Chomping-Down.txt Pacman-Chomping-Left.txt Pacman-Chomping-Right.txt Pacman-Chomping-Up.txt Pacman-Death.txt Pill.txt Red-Ghost-Down.txt Red-Ghost-Left.txt Red-Ghost-Right.txt Red-Ghost-Up.txt Red-Ghost-Win.txt Terrified-Ghost.txt
```
Comment (ForNeVeR) [inline sprites/Makefile]: I don't like the `TXTS` name very much. Maybe name it like `TXT_IMGS`? These are not ordinary text files, after all.

File: sprites/Makefile [hunk 2]
```diff
@@ -12,6 +16,9 @@ Pill.json: Pill.raw.json
 Pill.xpm: Pill.png
 	convert Pill.png Pill.xpm
 
+Pill.txt: Pill.png
+	convert Pill.png Pill.txt
```
Comment (ForNeVeR) [inline sprites/Makefile]: I am not expert on makefiles at all, but won't this rule create a collision with `%.txt`?

File: test/pacmacs-image-test.el [hunk 3]
```diff
@@ -95,3 +95,32 @@
...
+
+(ert-deftest pacmacs--parse-im-enum-header-test ()
+  (let* ((input-header "# ImageMagick pixel enumeration: 80,40,255,srgba")
+         (expected-result (list 80 40 255)))
+    (should (equal expected-result
+                   (pacmacs--parse-im-enum-header input-header)))
+    (should-error (pacmacs--parse-im-enum-header "khooy") :type 'error)))
+
+(ert-deftest pacmacs--parse-im-enum-pixel-test ()
+  (let ((input-pixel "60,16: (255,255,255,255)  #khooy")
```
Comment (ForNeVeR) [inline test/pacmacs-image-test.el]: I think that's better to add spaces here to match the original format closer:

```
(  255,  255,  255,  255)
```

So the test will fail if we disable a whitespace handling in our regexes somehow.

**Comment by ForNeVeR** [comment]
@rexim please evaluate if this PR will require to change our Melpa recipe.

File: sprites/Makefile (hunk 2 above)
Comment (rexim) [inline sprites/Makefile]: Good eye! It's actually a leftover of my experiments. I'll remove it. :)

File: sprites/Makefile (hunk 1 above)
Comment (rexim) [inline sprites/Makefile]: Agree. Will give a better name

File: test/pacmacs-image-test.el (hunk 3 above)
Comment (rexim) [inline test/pacmacs-image-test.el]: Affirmative

**Comment by rexim** [comment]
@ForNeVeR since #216 is a subtask of #41 (see https://github.com/codingteam/pacmacs.el/issues/41#issuecomment-180957182) we don't need to update MELPA recipe right now. But we definitely need to change it if the outcome of #41 will be positive. We need to keep track of such things somehow. I think I will create an issue for that.

**Comment by ForNeVeR** [comment]
Acknowledged.

**Comment by rexim** [comment]
@ForNeVeR fixed all the remarks. Please verify.

**Comment by rexim** [comment]
Sorry, assigning back to myself. Please see https://github.com/codingteam/pacmacs.el/issues/218#issuecomment-180975075. I need to change the Cask file to include those `sprites/*.txt` files. Let's keep track of the MELPA recipe changes in our Cask file.

**Comment by rexim** [comment]
@ForNeVeR done. Please, review now. :)

**Comment by ForNeVeR** [comment]
LGTM :+1:

**Comment by rexim** [comment]
Thank you Mr @ForNeVeR

**Pull request merged**